src/index/functions.rs

```diff
@@ -84,3 +86,26 @@ impl Drop for Index {
         }
     }
 }
+
+#[pgrx::pg_extern(sql = "")]
+fn _vchordrq_sampled_vectors(indexrelid: Oid) -> SetOfIterator<'static, String> {
+    let pg_am = PgAm::search_amname(c"vchordrq").unwrap();
+    let Some(pg_am) = pg_am.get() else {
+        pgrx::error!("vchord is not installed");
+    };
+    let pg_class = PgClass::search_reloid(indexrelid).unwrap();
+    let Some(pg_class) = pg_class.get() else {
+        pgrx::error!("the relation does not exist");
+    };
+    if pg_class.relkind() != PgClassRelkind::Index {
+        pgrx::error!("the relation {:?} is not an index", pg_class.relname());
+    }
+    if pg_class.relam() != pg_am.oid() {
+        pgrx::error!("the index {:?} is not a vchordrq index", pg_class.relname());
+    }
+    // The user must have access to the index, if not, raise an error from Postgres.
+    let _relation = Index::open(indexrelid, pgrx::pg_sys::AccessShareLock as _);
+    let max_records = crate::index::gucs::vchordrq_query_sampling_max_records();
```

Comment:
We should return all rows here. `max_records` will control the output too strangely.